preact@10.15.0 newly doesn't compile with TypeScript exactOptionalPropertyTypes

**Describe the bug**

`preact@10.14.1` compiles when TypeScript's [`exactOptionalPropertyTypes` tsconfig setting](https://www.typescriptlang.org/tsconfig#exactOptionalPropertyTypes) is enabled. `preact@10.15.0` doesn't:
```
node_modules/preact/compat/src/index.d.ts:140:19 - error TS2430: Interface 'ForwardRefExoticComponent<P>' incorrectly extends interface 'FunctionComponent<P>'.
  Types of property 'defaultProps' are incompatible.
    Type 'Partial<P> | undefined' is not assignable to type 'Partial<P>'.
      Type 'undefined' is not assignable to type 'Partial<P>'.

140  export interface ForwardRefExoticComponent<P>
                      ~~~~~~~~~~~~~~~~~~~~~~~~~
```

It's complaining that `FunctionComponent<P>` in src/index.d.ts has `defaultProps?: Partial<P>` but `ForwardRefExoticComponent<P>` in compat/src/index.d.ts (added at b55567f63dde28fb83f35e1615db9d3699c0de9c) has `defaultProps?: Partial<P> | undefined` instead of `defaultProps?: Partial<P>`.

**To Reproduce**

```shell
npm init --yes
npm install preact@10.15.0 typescript@5.0.4
echo "import 'preact/compat'" > index.ts
npx tsc index.ts --strictNullChecks --exactOptionalPropertyTypes
```

**Expected behavior**

Ideally, preact would continue to compile with this TypeScript configuration.

--- a/src/index.d.ts
+++ b/src/index.d.ts
@@ -89,7 +89,7 @@ export type ComponentProps<
 export interface FunctionComponent<P = {}> {
 	(props: RenderableProps<P>, context?: any): VNode<any> | null;
 	displayName?: string;
-	defaultProps?: Partial<P>;
+	defaultProps?: Partial<P> | undefined;
 }
 export interface FunctionalComponent<P = {}> extends FunctionComponent<P> {}
 

